helper/resource/testing_new_import_state.go

```diff
@@ -93,11 +93,35 @@ func testStepNewImportState(ctx context.Context, t testing.T, helper *plugintest
 
 	logging.HelperResourceTrace(ctx, fmt.Sprintf("Using import identifier: %s", importId))
 
-	// Create working directory for import tests
+	// Prepare the test config dependent on the kind of import test being performed
 	if testStepConfig == nil {
 		logging.HelperResourceTrace(ctx, "Using prior TestStep Config for import")
 
-		testStepConfig = cfg
+		switch step.ImportStateKind {
+		case ImportBlockWithResourceIdentity:
+			t.Fatalf("TODO implement me")
+		case ImportBlockWithId:
+			cfgRaw += fmt.Sprintf(`
+			import {
+				to = %s
+				id = %q
+			}
+		`, step.ResourceName, importId)
+		default:
+			// Not an import block test so nothing to do here
+		}
+
+		confRequest := teststep.PrepareConfigurationRequest{
+			Directory: step.ConfigDirectory,
+			File:      step.ConfigFile,
+			Raw:       cfgRaw,
+			TestStepConfigRequest: config.TestStepConfigRequest{
+				StepNumber: stepIndex + 1,
```

Comment:
Something I had overlooked in the proof-of-concept pull request: what is the significance of `stepIndex + 1` here (and at line 26)?

Reply:
I believe it's to maintain the correct step number which was modified to index starting from 1 for a better UX

https://github.com/hashicorp/terraform-plugin-testing/blob/7e59fd9813b7404cc5a55ad58a757f3b3bb030e5/helper/resource/testing_new.go#L144

This instance (as well as another in `testing_new.go`) can probably be updated to use `stepNumber`